src/nodestreamidguard.cpp

```diff
@@ -43,7 +43,9 @@ NodeStreamIdGuard::~NodeStreamIdGuard() {
         DECREMENT_IF_ENABLED(this->reporter.inferReqActive);
         this->inferRequestsQueue_.returnStream(this->streamId.value());
     }
-    DECREMENT_IF_ENABLED(this->reporter.currentRequests);
+    if (!disarmed) {
```

Comment:
Why do we check for disarmed twice? Line 36 and 46. We can simplify and move it inside first scope.